### DISCUSSION TITLE
Next.js middleware function not deployed?

### DISCUSSION THREAD

**Discussion by runoncedev** [vercel/community#754] [Help] [answered]
I'm trying to use Next.js middleware to do some redirects:

```
import { NextResponse } from "next/server";

export function middleware(request) {
  const matches = request.nextUrl.pathname.match(
    /^\/(?!maintenance|_next|favicon).*$/
  );

  if (matches) {
    return NextResponse.redirect(new URL("/maintenance", request.url));
  }

  return NextResponse.next();
}
```

If I run the build locally it's working fine but when deployed on Vercel it seems the middleware is not being executed and I can't find it under the "Functions" tab of my project.

Here is the repository with the middleware:

* https://github.com/oncet/gurim-next/blob/maintenance-mode/middleware.js

I already went through anything I could find online, including [this article from the support section](https://vercel.com/support/articles/why-does-my-serverless-function-work-locally-but-not-when-deployed).

What I'm missing?

**Comment by amyegan** [comment, marked as answer]
Thanks for sharing your repo and the steps you have already tried! I pulled down a copy of your code and tried running it locally but the middleware is not working for me ☹️ Do you have any config that is not being checked in from your local version? It seems like you might have something set up a bit differently on your local copy to make it work on locahost.

**Reply by amyegan** [reply]
I dug into it a little more and found something that works. Try moving your `middleware.js` into the `pages` directory and rename it as `_middleware.js`. 

You're using a slightly older version of Next.js so the setup needed to  be a little different. You can read more about the difference here: https://nextjs.org/docs/messages/middleware-upgrade-guide#explanation

**Reply by runoncedev** [reply]
Thank you!

There is something odd because locally I'm getting this result inside the project:

```
npm why next
next@12.2.3
```

But for some reason it doesn't seem to be reflected in the `package-lock.json`.

**Reply by runoncedev** [reply]
Manually bumped to v12.2 and now it's working fine, thank you for your time!
